include/ginkgo/core/base/executor.hpp

```diff
@@ -646,6 +649,97 @@ class Executor : public log::EnableLogging<Executor> {
         return this->verify_memory_from(other.get());
     }
 
+    /**
+     * A struct that abstracts the executor info for different executors
+     * classes.
+     */
+    struct exec_info {
+        /**
+         * The id of the device.
+         */
+        int device_id = -1;
+
+        /**
+         * The type of the device, relevant only for the dpcpp executor.
+         */
+        std::string device_type;
+
+        /**
+         * The numa node of the executor.
+         */
+        int numa_node = -1;
+
+        /**
+         * The number of computing units in the executor.
+         *
+         * @note In CPU executors this is equivalent to the number of cores.
+         *       In CUDA and HIP executors this is the number of Streaming
+         *       Multiprocessors. In DPCPP, this is the number of computing
+         *       units.
+         */
+        int num_computing_units = -1;
+
+        /**
+         * The number of processing elements per computing unit in the executor.
+         *
+         * @note In CPU executors this is equivalent to the number of SIMD units
+         *       per core. In CUDA and HIP executors this is the number of warps
+         *       per SM. In DPCPP, this is the max work group size.
+         */
+        int num_pe_per_cu = -1;
+
+        /**
+         * The sizes of the subgroup for the executor.
+         *
+         * @note In CPU executors this is invalid.
+         *       In CUDA and HIP executors warp size.
+         *       In DPCPP, this is the subgroup sizes for the device associated
+         *       with the dpcpp executor.
+         */
+        std::vector<int> subgroup_sizes{};
+
+        /**
+         * The sizes of the work items for the executor.
+         *
+         * @note In CPU executors this is invalid.
+         *       In CUDA and HIP executors this is invalid.
+         *       In DPCPP, this is the maximum sizes for the workitems for the
+         *       device associated with the dpcpp executor.
+         */
+        std::vector<int> max_workitem_sizes{};
```

Comment:
I am not too sure about this. @tcojean mentioned that it is possible that some dpcpp device might have a different value.

Reply:
I think I will leave this as a vector, because there are essentially no performance impacts as this is done during creation and as it is also easier to manage in case something in the future changes.